### PR TITLE
transforms: (set-memory-space) add support for streaming regions

### PR DESCRIPTION
**Pull request by jorendumoulin**
same as linalg.generics, force all memrefs to L1 for these operations

### CODE REVIEW

File: compiler/transforms/set_memory_space.py [hunk 1]
```diff
@@ -188,6 +191,41 @@ def get_cast_op(operand) -> None | memref.MemorySpaceCast:
...
+            ] = memref.MemorySpaceCast.from_type_and_target_space(
+                operand, memreftype, L1
+            )
+        rewriter.insert_op(tuple(memory_cast_ops.values()), InsertPoint.before(op))
+
+        # replace all operands to casted values
+        for i in range(len(op.operands)):
+            if op.operands[i] in operands_to_memory_cast:
+                op.operands[i] = memory_cast_ops[op.operands[i]].dest
+
```
Comment (JosseVanDelm) [inline compiler/transforms/set_memory_space.py:227]: Can you rewrite the InitLinalgMemorySpace to be the same like this?
Now these do the pretty much the same, but are written quite differently.
The linalg one reuses previous memory space casts and creates a new op, the way you've written this pass seems more efficient, and to me they look like they can share the exact same code?

**Review by JosseVanDelm — APPROVED** [review]
Okay, but now we have two very similar passes that could share more code.

File: compiler/transforms/set_memory_space.py (hunk 1 above)
Comment (jorendumoulin) [reply compiler/transforms/set_memory_space.py]: Okay! I combined them into one pattern!

Comment (JosseVanDelm) [reply compiler/transforms/set_memory_space.py]: Thank you!

**Pull request merged**